src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx

```diff
@@ -3,149 +3,169 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React, { useCallback, useEffect, useState } from 'react';
-import {
-  EuiSmallButton,
-  EuiCompressedFormRow,
-  EuiSpacer,
-  EuiFlexGroup,
-  EuiFlexItem,
-  EuiButtonIcon,
-  EuiCompressedComboBox,
-  EuiComboBoxOptionOption,
-  EuiFormLabel,
-} from '@elastic/eui';
+import React, { useState, useCallback } from 'react';
+import { EuiSpacer, EuiFlexItem, EuiSmallButton, EuiFlexGroup, EuiPanel } from '@elastic/eui';
 import { i18n } from '@osd/i18n';
-import { SavedObjectsStart } from '../../../../../core/public';
-import { getDataSourcesList } from '../../utils';
-import { DataSource } from '../../../common/types';
+import { SavedObjectsStart, CoreStart } from '../../../../../core/public';
+import { DataSource, DataSourceConnection, DataSourceConnectionType } from '../../../common/types';
 import { WorkspaceFormError } from './types';
+import { AssociationDataSourceModal } from '../workspace_detail/association_data_source_modal';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { WorkspaceClient } from '../../workspace_client';
+import { AssociationDataSourceModalMode } from '../../../common/constants';
+import { DataSourceConnectionTable } from '../workspace_detail/data_source_connection_table';
+import { DataSourceEngineType } from '../../../../data_source/common/data_sources';
 
 export interface SelectDataSourcePanelProps {
   errors?: { [key: number]: WorkspaceFormError };
   savedObjects: SavedObjectsStart;
-  selectedDataSources: DataSource[];
+  assignedDataSources: DataSource[];
   onChange: (value: DataSource[]) => void;
+  isDashboardAdmin: boolean;
 }
 
 export const SelectDataSourcePanel = ({
   errors,
```

Comment:
Seems `errors` is not used any more, is it by intention?

Reply:
no, I forgot to delete it